frontend/src/components/MapView/MapTooltip/index.tsx

```diff
@@ -117,6 +55,99 @@ const styles = () =>
     },
   });
 
-export interface TooltipProps extends WithStyles<typeof styles> {}
+const { multiCountry } = appConfig;
+const availableAdminLevels: AdminLevelType[] = multiCountry
+  ? [0, 1, 2]
+  : [1, 2];
+
+interface TooltipProps extends WithStyles<typeof styles> {}
+
+const MapTooltip = ({ classes }: TooltipProps) => {
+  const popup = useSelector(tooltipSelector);
+  const { i18n } = useSafeTranslation();
+  const [popupTitle, setPopupTitle] = useState<string>('');
+  const [adminLevel, setAdminLevel] = useState<AdminLevelType | undefined>(
+    undefined,
+  );
+
+  const { dataset, isLoading } = usePointDataChart();
+
+  const defaultPopupTitle = useMemo(() => {
+    if (isEnglishLanguageSelected(i18n)) {
+      return popup.locationName;
+    }
+    return popup.locationLocalName;
+  }, [i18n, popup.locationLocalName, popup.locationName]);
+
+  const renderedPopupLoader = useMemo(() => {
+    if (!popup.wmsGetFeatureInfoLoading) {
+      return null;
+    }
+    return <LinearProgress />;
+  }, [popup.wmsGetFeatureInfoLoading]);
+
+  const popupData = popup.data;
+
+  // TODO - simplify logic once we revamp admin levels ojbect
+  const adminLevelsNames = useCallback(() => {
+    const locationName = isEnglishLanguageSelected(i18n)
+      ? popup.locationName
+      : popup.locationLocalName;
+    const splitNames = locationName.split(', ');
+
+    const adminLevelLimit =
+      adminLevel === undefined
+        ? availableAdminLevels.length
+        : adminLevel + (multiCountry ? 1 : 0);
+    // If adminLevel is undefined, return the whole array
+    // eslint-disable-next-line fp/no-mutating-methods
```

Comment:
some of this logic seems related to RedirectToDMP only, could it be moved in there?